semeio/jobs/auto_correlate_observations/config.py

```diff
@@ -0,0 +1,315 @@
+import collections
+import configsuite as cs
+import copy
+import fnmatch
+
+
+_SCALING = "scaling"
+_THRESHOLD = "threshold"
+_STD_CUTOFF = "std_cutoff"
+_ALPHA = "alpha"
+_CLUSTERING = "clustering"
+_METHOD = "method"
+_SPEARMAN_CORRELATION = "spearman_correlation"
+_FCLUSTER = "fcluster"
+_T = "t"
+_CRITERION = "criterion"
+_INCONSISTENT = "inconsistent"
+_DISTANCE = "distance"
+_MAXCLUST = "maxclust"
+_MONOCRIT = "monocrit"
+_MAXCLUST_MONOCRIT = "maxclust_monocrit"
+_DEPTH = "depth"
+_LINKAGE = "linkage"
+_SINGLE = "single"
+_COMPLETE = "complete"
+_AVERAGE = "average"
+_WEIGHTED = "weighted"
+_CENTROID = "centroid"
+_WARD = "ward"
+_METRIC = "metric"
+_EUCLIDEAN = "euclidean"
+_METRICS = (
+    "braycurtis",
+    "canberra",
+    "chebyshev",
+    "cityblock",
+    "correlation",
+    "cosine",
+    "dice",
+    _EUCLIDEAN,
```

Comment:
it's quite many metrics, are all the metrics really necessary to be exposed? If yes I would perhaps suggest this (in order not to define name of each single metric):
```python
from enum import Enum

class Metrics(Enum):
         euclidean=0
         cityblock=1

# then you can access name
Metrics.euclidean.name
```

Reply:
In general I aimed for a very thin abstraction on top of `fcluster` and `linkage` such that the users can be guided to the `scipy` documentation and expect to find the same arguments exposed in the configuration. I could of course remove some of the metrics, but all that would save me was to define the strings... The values is propagated in the same manner no matter if we support only three of them or all.